packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -495,6 +497,81 @@ class _AnimatedPositionedState extends AnimatedWidgetBaseState<AnimatedPositione
   }
 }
 
+/// Animated widget that automatically transitions the child's size over a given
+/// duration whenever the given child's size changes.
+class AnimatedSize extends StatefulWidget {
+  /// Creates a widget that animates the size of its child implicitly.
+  ///
+  /// The [curve] and [duration] arguments must not be null.
+  AnimatedSize({
+    Key key,
+    this.child,
+    this.curve: Curves.linear,
+    @required this.duration
+  }) : super(key: key);
+
+  /// The child widget whose size change triggers the animation.
+  final Widget child;
+
+  /// The curve to apply when animating the parameters of this container.
+  final Curve curve;
+
+  /// The duration over which to animate the parameters of this container.
+  final Duration duration;
+
+  @override
+  _AnimatedSizeState createState() => new _AnimatedSizeState();
+}
+
+class _AnimatedSizeState extends State<AnimatedSize> {
+  Size _size;
+
+  @override
+  Widget build(BuildContext context) {
+    LayoutBuilder layoutBuilder = new LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        SchedulerBinding.instance.addPostFrameCallback(
+          (Duration timeStamp) {
+            context.visitChildElements(
+              (Element element) {
+                RenderBox renderBox = element.renderObject;
+
+                if (_size != renderBox.size) {
+                  setState(() {
+                    _size = renderBox.size;
```

Comment:
Hum...  Looks like an old widget we used to have called SizeObserver...  /cc @Hixie